pyro/infer/autoguide/guides.py

```diff
@@ -501,22 +506,23 @@ class AutoDiagonalNormal(AutoContinuous):
         svi = SVI(model, guide, ...)
 
     By default the mean vector is initialized to zero and the scale is
-    initialized to the identity.  To change this default behavior the user
-    should call :func:`pyro.param` before beginning inference, e.g.::
-
-        latent_dim = 10
-        pyro.param("auto_loc", torch.randn(latent_dim))
-        pyro.param("auto_scale", torch.ones(latent_dim),
-                   constraint=constraints.positive)
+    initialized to the identity.
     """
 
+    def __init__(self, model, prefix="auto", init_loc_fn=init_to_median,
+                 init_scale=1.0):
+        if not isinstance(init_scale, float) or not (init_scale > 0):
+            raise ValueError("Expected init_scale > 0. but got {}".format(init_scale))
```

Comment:
Is there a use case where we might want `init_scale` to be differently initialized per parameter? I think in that case we could also allow for a tensor of `latent_size` (though this can also be easily relaxed later depending on use)?

Reply:
Yes, we might want to do that. I think we can handle that in a subsequent PR, where we allow say `init_scale` to be a function like `init_loc_fn` and evaluate it on a per-site basis.

Reply:
Since these are parameters it may also be easier for the users to use the default initialization and rescale it later on a component-wise basis. I'm not fully sure if they will be able to do things like `self.scale[0] = ..` though.